### PR TITLE
Rewrite how rz_cons_pipe_open/close behave to remove globals

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Removed the old code because it was just wrong and replaced with the correct one.

### CODE REVIEW

**Review by XVilka — CHANGES_REQUESTED** [review]
Broken tests:
```
-- stderr
ERROR: cpipe: Cannot duplicate 8 to 1
WARNING: Could not open pipe to 1
ERROR: core: Error while executing command: aaa >/dev/null
```

**Comment by wargio** [comment]
> Broken tests:
> 
> ```
> -- stderr
> ERROR: cpipe: Cannot duplicate 8 to 1
> WARNING: Could not open pipe to 1
> ERROR: core: Error while executing command: aaa >/dev/null
> ```

i made a mistake an expected the same behaviour between windows and unix, but `dup2` behaves in a different way.

File: librz/cons/cpipe.c [hunk 1]
```diff
@@ -1,62 +1,138 @@
...
+
+#endif /* __WINDOWS__ */
+
+struct rz_cons_pipe_t {
+	int fd; ///< File descriptor number to override.
+	int copy_fd; ///< Copy of the file descriptor.
+	int file_fd; ///< File descriptor of the opened file.
+};
+
+RZ_API RZ_OWN RzConsPipe *rz_cons_pipe_open(RZ_NONNULL const char *file, int fd, bool append) {
```
Comment (XVilka) [inline librz/cons/cpipe.c:85]: Please add Doxygen

File: librz/cons/cpipe.c [hunk 2]
```diff
@@ -1,62 +1,138 @@
...
 
-RZ_API void rz_cons_pipe_close(int fd) {
-	if (fd != -1) {
-		close(fd);
-		if (backup_fd != -1) {
-			dup2(backup_fd, backup_fdn);
-			close(backup_fd);
-			backup_fd = -1;
-		}
+RZ_API void rz_cons_pipe_close(RZ_NULLABLE RzConsPipe *cpipe) {
```
Comment (XVilka) [inline librz/cons/cpipe.c:137]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]
Code is fine but please resend from the `dist-` branch to test all configurations

**Pull request closed without merging**